src/main/java/com/google/sps/data/restrictTimesCriteria.java

```diff
@@ -0,0 +1,54 @@
+package com.google.sps.data;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+public class restrictTimesCriteria implements Preference {
+
+  public restrictTimesCriteria(List<TimeRange> userNoClassTimes) {
+    this.userNoClassTimes = userNoClassTimes;
+  }
+
+  public List<TimeRange> userNoClassTimes;
+
+  /**
+   * Implements the preference score of the schedule, based on the criteria
+   * that prioritizes the TimeRanges that a user does not want class during
+   */
+  public float preferenceScore(Schedule schedule) {
+    // count the number of minutes that overlap
+    float minutesPenalty = 0;
+    
+    List<Course> courses = (List) schedule.getCourses();
+    List<TimeRange> timeOrderedCourses = getCourseTimes(courses);
+
+    for (TimeRange event : userNoClassTimes) {
+      for (TimeRange course : timeOrderedCourses) {
+        if (event.overlaps(course)) {
+          float timeDiff = (float) calculateDifference(event, course);
+          minutesPenalty -= timeDiff;
+        }
+      }
+    }
+    return minutesPenalty;
+  }
+
+  // Calculates the overlapping minutes between two TimeRanges
+  int calculateDifference(TimeRange a, TimeRange b) {
+    int start = Math.max(a.start(), b.start());
+    int end = Math.min(a.end(), b.end());
+    return end - start;
```

Comment:
This will always return a negative value (which i see that you took into account on line 32). It might be easier to have start be the Math.min and end be the Math.max, which is also in line with how TimeRanges deal with "earlier" and "later" times. (An earlier time is a lower number than a later time).

Reply:
I'm essentially trying to find the difference that `a` and `b` overlap by, such that I update the minutesPenalty with that difference. I keep it negative because the more overlap means the less desirable it is.